### DISCUSSION TITLE
Can we forward to different dns Upstream server with defferent protocol

### DISCUSSION THREAD

**Discussion by VDuyHoang** [TechnitiumSoftware/DnsServer#1126] [Q&A] [answered]
Hi,
Example like AGH we can define dns Upstream forwarder like this:
h3://...
quic://..
tls://...

**Comment by ShreyasZare** [comment, marked as answer]
Thanks for asking. Its not directly supported in the Settings GUI to prevent misconfig issues. You can do that by creating a root conditional forwarder zone by using `.` as the zone name and then add multiple FWD records which you can configure to use different protocols.

**Comment by VDuyHoang** [comment]
Yes, thank you. When i setup this the Advanced Forwarding APP doesn't work. it only use the root forward for dns lookup.
Another question when set root zone forward, does it use concurrent forwarding and how many Forwarder Concurrency it use ?

**Reply by ShreyasZare** [reply]
The zones on the DNS server are always given priority. So you should either use the root forwarder zone or configure the Advanced Forwarding app to use multiple different forwarders with different protocols. 

The Advanced Forwarding app is useful only when you need to have different forwarding for different client IP addresses. If you do not need such a feature then using the root forwarder zone will be sufficient.
